tools/reviewer/job/tasks/ReviewerMetadataUpdaterTask.java

```diff
@@ -153,5 +190,126 @@ public void run() {
+      ReviewerConfig reviewerConfig1, ReviewerConfig reviewerConfig2) {
+    final String displayName = reviewerConfig1.getDisplayName();
+    LinkedHashSet<Repo> repoList = new LinkedHashSet<>();
+    repoList.addAll(reviewerConfig1.getRepoList());
+    repoList.addAll(reviewerConfig2.getRepoList());
+    LinkedHashSet<Project> projectList = new LinkedHashSet<>();
+    projectList.addAll(reviewerConfig1.getProjectList());
+    projectList.addAll(reviewerConfig2.getProjectList());
+    LinkedHashSet<User> mergedUsersList = new LinkedHashSet<>();
+    for (User user1 : reviewerConfig1.getUserList()) {
+      User user2 = getUser(reviewerConfig2, user1.getId());
+      if (user2.getId().isEmpty()) {
+        // (user defined only in reviewerConfig1). Add `user1` to `mergedUsersList` without changes
+        mergedUsersList.add(user1);
+      } else {
+        // (users defined in both repos). Merge data and add merged result to `mergedUsersList`
+        String lastName = "";
+        String email = "";
+        String imageUrl = "";
+        if (!user1.getLastName().isEmpty()) {
+          lastName = user1.getLastName();
+        }
+        if (!user1.getEmail().isEmpty()) {
+          if (!user1.getEmail().equals(user2.getEmail())) {
+            System.out.println("***Emails for user " + user1.getId() + " differ between files.");
+          }
+          email = user1.getEmail();
+        }
+        if (!user1.getImageUrl().isEmpty()) {
+          if (!user1.getImageUrl().equals(user2.getImageUrl())) {
+            System.out.println(
+                "***Image Urls for user " + user1.getId() + " differ between files.");
+          }
+          imageUrl = user1.getImageUrl();
+        }
+        if (user1.getCrystals() != user2.getCrystals()) {
+          System.out.println(
+              "***Crystals amount for user " + user1.getId() + " differ between files.");
+        }
+        int crystals = 0;
+        crystals = user1.getCrystals();
+        LinkedHashSet<SocialNetwork> mergedUserSocialNetworks = new LinkedHashSet<>();
+        mergedUserSocialNetworks.addAll(user1.getSocialNetworkList());
+        mergedUserSocialNetworks.addAll(user2.getSocialNetworkList());
+        LinkedHashSet<String> mergedUserSkillList = new LinkedHashSet<>();
+        mergedUserSkillList.addAll(user1.getSkillList());
+        mergedUserSkillList.addAll(user2.getSkillList());
+        LinkedHashSet<String> mergedUserProjectIdList = new LinkedHashSet<>();
+        mergedUserProjectIdList.addAll(user1.getProjectIdList());
+        mergedUserProjectIdList.addAll(user2.getProjectIdList());
+        LinkedHashSet<Contribution> mergedUserContributions = new LinkedHashSet<>();
+        mergedUserContributions.addAll(user1.getTopContributionList());
+        mergedUserContributions.addAll(user2.getTopContributionList());
+        User.Builder mergedUserBuilder =
+            User.newBuilder()
+                .setId(user1.getId())
```

Comment:
Let's simplify by taking these values from `user1`:
`id`
`first_name`
`last_name`
`email`
`image_url`
`crystals`
`social_networks`
`skills`

So, no need to use intermediate variables. We can copy directly (like you do with `id`).
Alternatively, you can start with `user1.toBuilder()` and then you already have all the values (but you have to replace the ones you merge).

So, basically, we only need to merge these:
`project_id`
`top_contribution`

Reply:
Done.